contracts/StakeManager.sol

```diff
@@ -297,54 +318,131 @@ contract StakeManager is Ownable {
             pendingReward -= userReward;
             stakedToken.transfer(account.rewardAddress, userReward);
         }
-        if (userEpoch == currentEpoch && address(migration) == address(0)) {
-            mintMultiplier(account, block.timestamp, epochs[currentEpoch]);
+        mpDifference = account.currentMP - mpDifference;
+        if (address(migration) != address(0)) {
+            migration.increaseMPFromMigration(mpDifference);
+        } else if (userEpoch == currentEpoch) {
+            mintMP(account, block.timestamp, epochs[currentEpoch]);
         }
     }
 
-    function mintMultiplier(Account storage account, uint256 processTime, Epoch storage epoch) private {
-        uint256 deltaTime = processTime - account.lastMint;
-        account.lastMint = processTime;
-        uint256 increasedMultiplier = calcMaxMultiplierIncrease(
-            account.balance * ((MP_APY / YEAR) * deltaTime), account.multiplier, account.lockUntil, account.balance
-        );
-        account.multiplier += increasedMultiplier;
-        multiplierSupply += increasedMultiplier;
-        epoch.totalSupply += increasedMultiplier;
-        if (address(migration) != address(0)) {
-            migration.increaseMPFromMigration(increasedMultiplier);
+    /**
+     * @notice Mint initial multiplier points for given balance and time
+     * @dev if increased balance, increases difference of increased balance for current remaining lock time
+     * @dev if increased lock time, increases difference of total new balance for increased lock time
+     * @param account Account to mint multiplier points
+     * @param increasedLockTime increased lock time
+     * @param increasedBalance increased balance
+     */
+    function mintIntialMP(Account storage account, uint256 increasedLockTime, uint256 increasedBalance) private {
+        uint256 increasedMP;
+        if (increasedBalance > 0) {
+            increasedMP += increasedBalance; //initial multiplier points
+            if (block.timestamp < account.lockUntil) {
+                //increasing balance on locked account?
+                //bonus for remaining previously locked time of new balance.
+                increasedMP += getIncreasedMP(increasedBalance, account.lockUntil - block.timestamp);
+            }
         }
+        if (increasedLockTime > 0) {
+            //bonus for increased lock time
+            increasedMP += getIncreasedMP(account.balance + increasedBalance, increasedLockTime);
+        }
+
+        //does not check for MAX_BOOST
```

Comment:
Is this a TODO?

Reply:
Nope

Reply:
It just informs that it does not, and should not, check for max boost, because initial (bonus) multipliers are not part of this check. This is a bug that we had in the old code.